### ISSUE TITLE
Debug Spam with new JEI version 10 alpha

### ISSUE THREAD

**Issue by Ludi87** [SlimeKnights/TinkersConstruct#4915]
### Minecraft Version

1.18.2

### Forge Version

40.1.51

### Mantle Version

1.18.2-1.9.27

### Tinkers' Construct Version

1.18.2-3.5.1.31

### Describe your issue

version jei-1.18.2-forge-10.1.0.208 (alpha)

Lots of spam in the debug.log from a change related to fluids
apparently FLUID is FLUID_STACK now.

```
java.lang.NoSuchFieldError: FLUID
	at slimeknights.tconstruct.plugin.jei.AlloyRecipeCategory.drawVariableFluids(AlloyRecipeCategory.java:139) ~[TConstruct-1.18.2-3.5.1.31.jar%2354!/:1.18.2-3.5.1.31]
	at slimeknights.tconstruct.plugin.jei.melting.FoundryCategory.setRecipe(FoundryCategory.java:75) ~[TConstruct-1.18.2-3.5.1.31.jar%2354!/:1.18.2-3.5.1.31]
	at slimeknights.tconstruct.plugin.jei.melting.FoundryCategory.setRecipe(FoundryCategory.java:30) ~[TConstruct-1.18.2-3.5.1.31.jar%2354!/:1.18.2-3.5.1.31]
	at mezz.jei.common.util.IngredientSupplierHelper.getIngredientSupplier(IngredientSupplierHelper.java:24) ~[jei-1.18.2-forge-10.1.0.208.jar%2352!/:10.1.0.208]
	at mezz.jei.common.recipes.RecipeManagerInternal.lambda$addRecipes$0(RecipeManagerInternal.java:121) ~[jei-1.18.2-forge-10.1.0.208.jar%2352!/:10.1.0.208]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:178) ~[?:?]
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260) ~[?:?]
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616) ~[?:?]
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622) ~[?:?]
	at java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627) ~[?:?]
	at mezz.jei.common.recipes.RecipeManagerInternal.addRecipes(RecipeManagerInternal.java:127) ~[jei-1.18.2-forge-10.1.0.208.jar%2352!/:10.1.0.208]
	at mezz.jei.common.recipes.RecipeManagerInternal.addRecipes(RecipeManagerInternal.java:109) ~[jei-1.18.2-forge-10.1.0.208.jar%2352!/:10.1.0.208]
	at mezz.jei.common.load.registration.RecipeRegistration.addRecipes(RecipeRegistration.java:80) ~[jei-1.18.2-forge-10.1.0.208.jar%2352!/:10.1.0.208]
	at slimeknights.tconstruct.plugin.jei.JEIPlugin.registerRecipes(JEIPlugin.java:167) ~[TConstruct-1.18.2-3.5.1.31.jar%2354!/:1.18.2-3.5.1.31]
	at mezz.jei.common.load.PluginLoader.lambda$createRecipeManager$9(PluginLoader.java:188) ~[jei-1.18.2-forge-10.1.0.208.jar%2352!/:10.1.0.208]
	at mezz.jei.common.load.PluginCaller.callOnPlugins(PluginCaller.java:25) ~[jei-1.18.2-forge-10.1.0.208.jar%2352!/:10.1.0.208]
	at mezz.jei.common.load.PluginLoader.createRecipeManager(PluginLoader.java:188) ~[jei-1.18.2-forge-10.1.0.208.jar%2352!/:10.1.0.208]
	at mezz.jei.common.startup.JeiStarter.start(JeiStarter.java:86) ~[jei-1.18.2-forge-10.1.0.208.jar%2352!/:10.1.0.208]
	at mezz.jei.forge.startup.ClientLifecycleHandler.startJei(ClientLifecycleHandler.java:84) ~[jei-1.18.2-forge-10.1.0.208.jar%2352!/:10.1.0.208]
	at mezz.jei.forge.startup.StartEventObserver.transitionState(StartEventObserver.java:137) ~[jei-1.18.2-forge-10.1.0.208.jar%2352!/:10.1.0.208]
	at mezz.jei.forge.startup.StartEventObserver.onEvent(StartEventObserver.java:100) ~[jei-1.18.2-forge-10.1.0.208.jar%2352!/:10.1.0.208]
	at net.minecraftforge.eventbus.EventBus.doCastFilter(EventBus.java:247) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.eventbus.EventBus.lambda$addListener$11(EventBus.java:239) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-5.0.3.jar%232!/:?]
	at net.minecraft.client.multiplayer.ClientPacketListener.m_5859_(ClientPacketListener.java:1270) ~[client-1.18.2-20220404.173914-srg.jar%2355!/:?]
	at net.minecraft.network.protocol.game.ClientboundUpdateTagsPacket.m_5797_(ClientboundUpdateTagsPacket.java:35) ~[client-1.18.2-20220404.173914-srg.jar%2355!/:?]
	at net.minecraft.network.protocol.game.ClientboundUpdateTagsPacket.m_5797_(ClientboundUpdateTagsPacket.java:11) ~[client-1.18.2-20220404.173914-srg.jar%2355!/:?]
	at net.minecraft.network.protocol.PacketUtils.m_131356_(PacketUtils.java:22) ~[client-1.18.2-20220404.173914-srg.jar%2355!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_6367_(BlockableEventLoop.java:157) ~[client-1.18.2-20220404.173914-srg.jar%2355!/:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.m_6367_(ReentrantBlockableEventLoop.java:23) ~[client-1.18.2-20220404.173914-srg.jar%2355!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_7245_(BlockableEventLoop.java:131) ~[client-1.18.2-20220404.173914-srg.jar%2355!/:?]
	at net.minecraft.util.thread.BlockableEventLoop.m_18699_(BlockableEventLoop.java:116) ~[client-1.18.2-20220404.173914-srg.jar%2355!/:?]
	at net.minecraft.client.Minecraft.m_91383_(Minecraft.java:1013) ~[client-1.18.2-20220404.173914-srg.jar%2355!/:?]
	at net.minecraft.client.Minecraft.m_91374_(Minecraft.java:663) ~[client-1.18.2-20220404.173914-srg.jar%2355!/:?]
	at net.minecraft.client.main.Main.main(Main.java:205) ~[client-1.18.2-20220404.173914-srg.jar%2355!/:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.minecraftforge.fml.loading.targets.CommonClientLaunchHandler.lambda$launchService$0(CommonClientLaunchHandler.java:31) ~[fmlloader-1.18.2-40.1.51.jar%2316!/:?]
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:106) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:77) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) [modlauncher-9.1.3.jar%235!/:?]
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:149) [bootstraplauncher-1.0.0.jar:?]
[19Jun2022 02:29:32.163] [Render thread/ERROR] [mezz.jei.common.util.IngredientSupplierHelper/]: Found a broken recipe, failed to setRecipe with RecipeLayoutBuilder: Tinkers' Construct tconstruct:tools/materials/melting/manyullyn class slimeknights.tconstruct.library.recipe.melting.MeltingRecipe
```

### Other mods

jei-1.18.2-forge-10.1.0.208.jar
Mantle-1.18.2-1.9.27.jar
TConstruct-1.18.2-3.5.1.31.jar

### Tried reproducing with just Tinkers?

Yes

### Performance Enchancers

None of the above

### Searched for known issues?

Searched open issues

**Comment by Ludi87** [comment]
for more info, @mezz - I'm not a moddev, just encountered a similar issue with Thermal Expansion.

**Comment by Ludi87** [comment]
[latest.log](https://github.com/SlimeKnights/TinkersConstruct/files/8934559/latest.log)

**Comment by ffuentesm** [comment]
You need to roll back to a previous version of JEI.
Currently, JEI is trying to follow the new Fluid API that is in Forge 1.19 and they applied those changes to the new 1.18.2 JEI updates. So using a prior JEI version might fix the problem

**Comment by Ludi87** [comment]
Oh I don't need a fix, just wanted to let the tinkers devs know that something changed. But thanks for the info in case someone stumbles over this :)

**Comment by mooliberry** [comment]
I've come across the same issue. Tinkers' and JEI works fine together in singleplayer, but Tinkers' seems to stop JEI from loading recipes when starting the server. This is after 50/50 mods to find the culprit.

**Comment by Ludi87** [comment]
Yea as ffuentesm said, its probably best for now to stay on the JEI version 9 until most mods have adjusted to the changes

**Comment by mezz** [comment]
JEI 10 supports Fabric, so I had to make a breaking change and remove the deprecated `FLUID` field here.
If Tinker's updates their JEI integration to support JEI version 9.7 and move away from deprecated things, it'll work for JEI 10 as well.

**Comment by KnightMiner** [comment]
Based on that information, JEI version 9 should work with tinkers until we get a chance to update, so if you are experiencing this bug rollback JEI. It sounds like other mods can supper both JEI 9 and 10 so nothing else should break (ideally) from the older build.

**Comment by nexitem** [comment]
Rolling back to the current "Stable" version of JEI has solved this for me aswell.

**Comment by eman19977** [comment]
rolling back to a previous version causes my game to crash everytime

**Comment by KnightMiner** [comment]
Fixed by #4980

**Comment by KnightMiner** [comment]
3.4.2.40 has been updated to use the latest APIs, so there should be no further issues. If there are please open a new issue.
